Sync Merge: update-workshop-to-c5 (#797)
espressif/developer-portal: content/workshops/esp-idf-with-esp32-c5/assignment-1-1/index.md

* __Split the SSIDs on your router.__ Log into your router's admin page and give the 5 GHz radio its own SSID (for example, `MyNetwork_5G`). Most routers expose this under the wireless settings, sometimes behind an "advanced" or "separate bands" toggle. You then connect the board to that dedicated 5 GHz SSID.

* __Use a phone hotspot set to 5 GHz.__ Many modern phones let you choose the hotspot band. On Android, open the hotspot settings and select the __5 GHz__ band (sometimes labeled "Maximize compatibility" for 2.4 GHz versus a 5 GHz option). This is the quickest way to get a known 5 GHz network for testing.

* __Use a second board as a 5 GHz SoftAP.__ The `ESP32-C5` can act as an access point on the 5 GHz band. If you have a spare board, you can start a SoftAP on a non-DFS channel such as 36, 40, 44, or 48 and connect your main board to it.

Whichever option you choose, note down the __SSID__ and __password__ of the 5 GHz network. You'll plug them into the code in the next steps.

## Create a new project and add the Wi-Fi includes

> 1. Create a new ESP-IDF project.

Reviewer: Add the `idf.py` command to create a new project via CLI and the steps using the VSCode extension.

Reviewer 2: Added both as tabs

                                              pdFALSE,
                                              portMAX_DELAY);
   ```

2. If the connection succeeded, confirm the active band with `esp_wifi_get_band()` and cross-check it against the channel reported by `esp_wifi_sta_get_ap_info()`. With `WIFI_BAND_MODE_5G_ONLY` the band should always be 5 GHz, and channels of 36 and above confirm it:

   ```c
       if (bits & WIFI_CONNECTED_BIT) {
           // Check which band was used
           wifi_band_t band;
           ESP_ERROR_CHECK(esp_wifi_get_band(&band));

           if (band == WIFI_BAND_5G) {
               ESP_LOGI(TAG, "Connected on 5 GHz!");
           } else {
               ESP_LOGI(TAG, "Connected on 2.4 GHz.");
           }

           // Also print the channel as a cross-check (5 GHz channels are 36+)
           wifi_ap_record_t ap_info;
           ESP_ERROR_CHECK(esp_wifi_sta_get_ap_info(&ap_info));
           ESP_LOGI(TAG, "Connected to SSID: %s, Channel: %d, RSSI: %d",
                    ap_info.ssid, ap_info.primary, ap_info.rssi);

       } else {
           ESP_LOGI(TAG, "Failed to connect.");
       }
   ```

## Build, flash, and monitor

> 1. Build and flash the project: `> ESP-IDF: Build, Flash and Start a Monitor on your Device`.

Reviewer: Add CLI commands.

Reviewer 2: added

---
title: "ESP-IDF C5 - Assign. 1.1"
date: "2026-07-29"
lastmod: "2026-07-29"
series: ["WS00C"]
series_order: 3
showAuthor: false
summary: "Write an application that connects the ESP32-C5 to a Wi-Fi network as a station, restricts the radio to the 5 GHz band, and reports the band and channel it joined."
---

In this assignment, you'll put the theory from [Lecture 1](../lecture-1/) into practice. You'll bring up the Wi-Fi driver in station mode, connect to an access point, restrict the radio to the 5 GHz band with `esp_wifi_set_band_mode()`, and then read back the band the connection landed on.

## Assignment steps

1. Prepare a 5 GHz network to connect to.
2. Create a new project and add the Wi-Fi includes.
3. Handle the Wi-Fi and IP events.
4. Initialize NVS, the network stack, and the Wi-Fi driver.
5. Configure the station and select the band mode.
6. Wait for the connection and report the band.
7. Build, flash, and monitor.

## Prepare a 5 GHz network to connect to

Before writing any code, you need a 5 GHz access point that the `ESP32-C5` can join. This is worth a moment of attention, because most networks make the 5 GHz band harder to reach than you might expect.

> Many home routers broadcast the 2.4 GHz and 5 GHz radios under the __same SSID__ (a feature often called band steering or Smart Connect). When both bands share one name, a station only sees a single network. Because this assignment sets the band mode to `WIFI_BAND_MODE_5G_ONLY`, the board will only ever join the 5 GHz radio, so a shared SSID is fine as long as the access point actually broadcasts on 5 GHz. Any of the options below gives you a network you can be sure supports the 5 GHz band.

Reviewer 3: ```suggestion
Many home routers broadcast the 2.4 GHz and 5 GHz radios under the __same SSID__ (a feature often called band steering or Smart Connect). When both bands share one name, a station only sees a single network. Because this assignment sets the band mode to `WIFI_BAND_MODE_5G_ONLY`, the board will only ever join the 5 GHz radio, so a shared SSID is fine as long as the access point actually broadcasts on 5 GHz. Here are the options to have a network that supports the 5 GHz band:
```

       ESP_ERROR_CHECK(esp_wifi_set_config(WIFI_IF_STA, &wifi_config));
       ESP_ERROR_CHECK(esp_wifi_start());
   ```

2. Set the band mode to `WIFI_BAND_MODE_5G_ONLY` so the driver restricts the radio to the 5 GHz band and only connects to 5 GHz access points:

   ```c
       // Restrict Wi-Fi to the 5 GHz band only.
       // Must be called after esp_wifi_start(), otherwise it returns
       // ESP_ERR_WIFI_NOT_STARTED (0x3002).
       ESP_ERROR_CHECK(esp_wifi_set_band_mode(WIFI_BAND_MODE_5G_ONLY));
   ```

   > [!IMPORTANT]
   > `esp_wifi_set_band_mode()` must be called __after__ `esp_wifi_start()`. If you call it earlier, it returns `ESP_ERR_WIFI_NOT_STARTED` (`0x3002`) and `ESP_ERROR_CHECK` will abort the program.

   > [!TIP]
   > Here you force the board onto 5 GHz with `WIFI_BAND_MODE_5G_ONLY`. If you would rather let the board fall back to 2.4 GHz when 5 GHz is unavailable, use `WIFI_BAND_MODE_AUTO` instead, which scans both bands and picks the strongest match. In `AUTO` mode you can bias the choice toward 5 GHz using the `rssi_5g_adjustment` field of the scan threshold.

## Wait for the connection and report the band

> Finally, block until the event handler signals success or failure, then read back which band the board actually joined.

Reviewer 3: > block until ...

What should be blocked?

1. Wait on the event group for either the connected bit or the fail bit:

   ```c
       // Wait for connection or failure
       EventBits_t bits = xEventGroupWaitBits(s_wifi_event_group,
                                              WIFI_CONNECTED_BIT | WIFI_FAIL_BIT,
                                              pdFALSE,
                                              pdFALSE,
                                              portMAX_DELAY);
   ```

> 2. If the connection succeeded, confirm the active band with `esp_wifi_get_band()` and cross-check it against the channel reported by `esp_wifi_sta_get_ap_info()`. With `WIFI_BAND_MODE_5G_ONLY` the band should always be 5 GHz, and channels of 36 and above confirm it:

Reviewer 3: > and channels of 36 and above confirm it

Do I understand it correctly that channels 36 and above confirm that the band is always be 5 GHz?

   ```c
       if (bits & WIFI_CONNECTED_BIT) {
           // Check which band was used
           wifi_band_t band;
           ESP_ERROR_CHECK(esp_wifi_get_band(&band));

           if (band == WIFI_BAND_5G) {
               ESP_LOGI(TAG, "Connected on 5 GHz!");
           } else {
               ESP_LOGI(TAG, "Connected on 2.4 GHz.");
           }

           // Also print the channel as a cross-check (5 GHz channels are 36+)
           wifi_ap_record_t ap_info;
           ESP_ERROR_CHECK(esp_wifi_sta_get_ap_info(&ap_info));
           ESP_LOGI(TAG, "Connected to SSID: %s, Channel: %d, RSSI: %d",
                    ap_info.ssid, ap_info.primary, ap_info.rssi);

       } else {
           ESP_LOGI(TAG, "Failed to connect.");
       }
   ```

## Build, flash, and monitor

1. Build and flash the project: 
    {{< tabs group="config" >}}
    {{< tab label="ESP-IDF Extension for VSCode" >}}
    `> ESP-IDF: Build, Flash and Start a Monitor on your Device`.
    
    {{< /tab >}}

    {{< tab label="CLI" >}}

    ```terminal
        idf.py -p PORT flash monitor
    ```     

>     where PORT is the port at which your device is attached. Then name depend on your O.S. (usually "COM" on Windows, "ttyusb" on Linux, "tty.usbserial" on macos).

Reviewer 3: ```suggestion
    Where `PORT` is the port on which your device is connected. The name depend on your OS (usually, Windows -- "COM", Linux -- "ttyusb", macOS -- "tty.usbserial").
```
